app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddCyclewayPartSurface.kt

```diff
@@ -29,7 +37,17 @@ class AddCyclewayPartSurface(o: OverpassMapDataAndGeometryApi, r: ResurveyInterv
 
     override fun createForm() = AddPathSurfaceForm()
 
-    override fun applyAnswerTo(answer: String, changes: StringMapChangesBuilder) {
-        changes.updateWithCheckDate("cycleway:surface", answer)
+    override fun applyAnswerTo(answer: DetailSurfaceAnswer, changes: StringMapChangesBuilder) {
+        when (answer) {
+            is SurfaceAnswer -> {
+                changes.updateWithCheckDate("cycleway:surface", answer.value)
+                changes.deleteIfExists("source:cycleway:surface")
+                changes.deleteIfExists("cycleway:surface:note")
+            }
+            is DetailingWhyOnlyGeneric -> {
```

Comment:
I think the naming of the classes is a bit off. I think it should be:

The interface/base class: `SurfaceAnswer`
The class that just includes the surface because it is already a detailed surface answer: `DetailSurfaceAnswer`
The class that includes both the description and the surface value: `GenericSurfaceAnswer` because the answer does not only contain the "detailing" but also the generic surface.

Reply:
I used not `DetailSurfaceAnswer` but `SpecificSurfaceAnswer` as I think that it is more, well, specific, how it differs from `GenericSurfaceAnswer`.

`DetailSurfaceAnswer` could be detailed in some other way.

Reply:
ok